lib/Wikia/src/Service/Gateway/GatewayBase.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Wikia\Service\Gateway;
+
+use Http;
+
+class GatewayBase {
+
+	private $urlProvider;
+
+	function __construct( UrlProviderInterface $urlProvider ) {
+		$this->urlProvider = $urlProvider;
+	}
+
+	public function get( $path, $headers = null ) {
+		return $this->doRequest( "GET", $path, $headers );
+	}
+
+	public function put( $path, $headers = null, $data = null ) {
+		return $this->doRequest( "PUT", $path, $headers, $data );
+	}
+
+	public function post( $path, $headers = null, $data = null ) {
+		return $this->doRequest( "POST", $path, $headers, $data );
+	}
+
+	public function delete( $path, $headers = null ) {
+		return $this->doRequest( "DELETE", $path, $headers );
+	}
+
+	private function doRequest( $method, $path, $headers = null, $data = null ) {
+		$url = $this->getUrl( $path );
+		$options = $this->getHttpRequestOption( $headers, $data );
+		$response = Http::Request( $method, $url, $options );
```

Comment:
We should inject the HTTP client so we don't have a dependency on MW in `/lib/Wikia`.